src/confluent_kafka/src/Producer.c

```diff
@@ -393,6 +393,83 @@ static PyObject *Producer_flush (Handle *self, PyObject *args,
         return cfl_PyInt_FromInt(qlen);
 }
 
+static PyObject *Producer_topic_new (Handle *self, PyObject *args,
+                                 PyObject *kwargs) {
+        rd_kafka_topic_conf_t *topic_conf = NULL;
+        rd_kafka_topic_t *topic_obj = NULL;
+        const char *topic = NULL;
+        PyObject *conf = NULL;
+        Py_ssize_t pos = 0;
+        PyObject *ko = NULL, *vo = NULL;
+
+        static char *kws[] = { "topic", "conf", NULL };
+
+        if (!PyArg_ParseTupleAndKeywords(args, kwargs, "s|O", kws, &topic, &conf))
+            return NULL;
+
+	    if (!PyDict_Check(conf))
+            return NULL;
+
+        topic_conf = rd_kafka_topic_conf_new();
+        while (PyDict_Next(conf, &pos, &ko, &vo)) {
+            PyObject *ks, *ks8 = NULL;
+            PyObject *vs = NULL, *vs8 = NULL;
+            const char *k = NULL, *v = NULL;
+            char errstr[256];
+
+            if (!(ks = cfl_PyObject_Unistr(ko))) {
+                PyErr_SetString(PyExc_TypeError,
+                                "expected configuration property name "
+                                "as type unicode string");
+                goto inner_err;
+            }
+
+            k = cfl_PyUnistr_AsUTF8(ks, &ks8);
```

Comment:
OMG... this wrapper to support py2 is ugly. I wish they could deprecate py2 and clean a lot of the internal helpers. :/